diff --git a/qemu_test.go b/qemu_test.go
index ae10471..cb7d31b 100644
--- a/qemu_test.go
+++ b/qemu_test.go
@@ -20,28 +20,8 @@ func init() {
 
 // detectLinuxKernel returns path to kernel/initramfs at the current system
 func detectLinuxKernel() (string, string, error) {
-	if _, err := os.Stat("/boot/vmlinuz-linux"); err == nil {
-		// it looks like Arch Linux
-		return "/boot/vmlinuz-linux", "/boot/initramfs-linux.img", nil
-	}
-
-	// Check if it is Debian?
-	var uts unix.Utsname
-
-	if err := unix.Uname(&uts); err != nil {
-		return "", "", fmt.Errorf("uname: %v", err)
-	}
-	length := bytes.IndexByte(uts.Release[:], 0)
-	version := string(uts.Release[:length])
-
-	kernel := fmt.Sprintf("/boot/vmlinuz-%v", version)
-	initram := fmt.Sprintf("/boot/initrd.img-%v", version)
-
-	if _, err := os.Stat(kernel); err != nil {
-		return "", "", fmt.Errorf("Cannot find Linux kernel file at this system")
-	}
-
-	return kernel, initram, nil
+	return "/run/current-system/kernel/bzImage",
+	"/run/current-system/initrd", nil
 }
 
 func TestBootCurrentLinuxKernelInQemu(t *testing.T) {
