--- a/doc/manual/local.mk	2022-10-30 18:52:44.091425741 +0100
+++ b/doc/manual/local.mk	2022-10-30 18:53:00.026549841 +0100
@@ -77,7 +77,6 @@
 
 # Generate the HTML manual.
 html: $(docdir)/manual/index.html
-install: $(docdir)/manual/index.html
 
 # Generate 'nix' manpages.
 install: $(mandir)/man1/nix3-manpages
@@ -101,7 +100,5 @@
 	done
 	@touch $@
 
-$(docdir)/manual/index.html: $(MANUAL_SRCS) $(d)/book.toml $(d)/anchors.jq $(d)/custom.css $(d)/src/SUMMARY.md $(d)/src/command-ref/new-cli $(d)/src/command-ref/conf-file.md $(d)/src/language/builtins.md
-	$(trace-gen) RUST_LOG=warn mdbook build doc/manual -d $(DESTDIR)$(docdir)/manual
 
 endif
